vavr/src/main/java/io/vavr/collection/Vector.java

```diff
@@ -1078,11 +1078,8 @@ public int lastIndexOf(T element, int end) {
 
     @Override
     public Vector<T> subSequence(int beginIndex, int endIndex) {
-        if ((beginIndex >= 0) && (beginIndex <= endIndex) && (endIndex <= length())) {
-            return slice(beginIndex, endIndex);
-        } else {
-            throw new IndexOutOfBoundsException("subSequence(" + beginIndex + ", " + endIndex + ") on Vector of size " + length());
-        }
+        Collections.subSequenceRangeCheck(beginIndex, endIndex, length());
+        return slice(beginIndex, endIndex);
```

Comment:
I know that we do not pull the last bit of performance out of the JVM but it is more robust and removes redundant code.